### PR TITLE
MULE-10602: Parameter Model should include groups and exclusive optio…

### PR DESCRIPTION
**Pull request by marianogonzalez**
…nals information

### CODE REVIEW

File: src/main/java/org/mule/runtime/api/meta/model/declaration/fluent/ParameterGroupDeclaration.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+  /**
+   * Adds a {@link ParameterDeclaration}
+   *
+   * @param parameter a not {@code null} {@link ParameterDeclaration}
+   * @return this declaration
+   * @throws {@link IllegalArgumentException} if {@code parameter} is {@code null}
+   */
+  public ParameterGroupDeclaration addParameter(ParameterDeclaration parameter) {
+    if (parameter == null) {
```
Comment (pablomusumeci) [inline src/main/java/org/mule/runtime/api/meta/model/declaration/fluent/ParameterGroupDeclaration.java]: https://github.com/mulesoft/mule-api/blob/master/src/main/java/org/mule/runtime/api/util/Preconditions.java#L21

File: src/main/java/org/mule/runtime/api/meta/model/declaration/fluent/ParameterizedDeclaration.java [hunk 2]
```diff
@@ -28,26 +34,52 @@
...
-   * @param parameter a not {@code null} {@link ParameterDeclaration}
+   * @param parameterGroup a not {@code null} {@link ParameterDeclaration}
    * @return this declaration
    * @throws {@link IllegalArgumentException} if {@code parameter} is {@code null}
    */
-  public T addParameter(ParameterDeclaration parameter) {
-    if (parameter == null) {
-      throw new IllegalArgumentException("Can't add a null parameter");
+  public T addParameterGroup(ParameterGroupDeclaration parameterGroup) {
+    if (parameterGroup == null) {
```
Comment (pablomusumeci) [inline src/main/java/org/mule/runtime/api/meta/model/declaration/fluent/ParameterizedDeclaration.java]: https://github.com/mulesoft/mule-api/blob/master/src/main/java/org/mule/runtime/api/util/Preconditions.java#L21

File: src/main/java/org/mule/runtime/api/meta/model/declaration/fluent/ParameterizedDeclaration.java [hunk 3]
```diff
@@ -28,26 +34,52 @@
...
+  public List<ParameterGroupDeclaration> getParameterGroups() {
+    return unmodifiableList(new ArrayList<>(parameterGroups.values()));
+  }
+
+  /**
+   * @param groupName the name of the group which declaration you seek
+   * @return the {@link ParameterGroupDeclaration} of the given {@code groupName}.
+   */
+  public ParameterGroupDeclaration getParameterGroup(String groupName) {
+    if (isBlank(groupName)) {
```
Comment (pablomusumeci) [inline src/main/java/org/mule/runtime/api/meta/model/declaration/fluent/ParameterizedDeclaration.java]: https://github.com/mulesoft/mule-api/blob/master/src/main/java/org/mule/runtime/api/util/Preconditions.java#L21

Comment (marianogonzalez) [inline src/main/java/org/mule/runtime/api/meta/model/declaration/fluent/ParameterizedDeclaration.java]: fixed

File: src/main/java/org/mule/runtime/api/meta/model/declaration/fluent/ParameterGroupDeclaration.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  /**
+   * Adds a {@link ParameterDeclaration}
+   *
+   * @param parameter a not {@code null} {@link ParameterDeclaration}
+   * @return this declaration
+   * @throws {@link IllegalArgumentException} if {@code parameter} is {@code null}
+   */
+  public ParameterGroupDeclaration addParameter(ParameterDeclaration parameter) {
+    checkArgument(parameter != null, "Can't add a null parameter");
+
```
Comment (pablomusumeci) [inline src/main/java/org/mule/runtime/api/meta/model/declaration/fluent/ParameterGroupDeclaration.java]: remove empty line

File: src/main/java/org/mule/runtime/api/meta/model/parameter/ParameterGroupModel.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+   * Group name for parameters that are considered for general purposes and shouldn't belong to a particular
+   * group.
+   */
+  String DEFAULT_GROUP_NAME = "General";
+
+  /**
+   * Group name for parameters that are considered for advanced usage.
+   */
+  String ADVANCED = "Advanced";
+  /**
```
Comment (canoasan) [inline src/main/java/org/mule/runtime/api/meta/model/parameter/ParameterGroupModel.java]: add an empty line.

File: src/main/java/org/mule/runtime/api/meta/model/declaration/fluent/ParameterGroupDeclaration.java (hunk 4 above)
Comment (marianogonzalez) [inline src/main/java/org/mule/runtime/api/meta/model/declaration/fluent/ParameterGroupDeclaration.java]: that line is separating the validation part of the method from the actual logic. We always do that usually. You should pay more attention to our code base

File: src/main/java/org/mule/runtime/api/meta/model/parameter/ParameterGroupModel.java (hunk 5 above)
Comment (marianogonzalez) [inline src/main/java/org/mule/runtime/api/meta/model/parameter/ParameterGroupModel.java]: great catch!

**Pull request merged**